opentech/apply/funds/templatetags/submission_tags.py

```diff
@@ -0,0 +1,27 @@
+import re
+
+from django import template
+from django.utils.safestring import mark_safe
+
+from opentech.apply.funds.models import ApplicationSubmission
+
+register = template.Library()
+
+
+@register.filter
+def submission_links(value):
+    regex = re.compile('[^\w]\#(\d+)[^\w]')
+    links = {}
+    for match in regex.finditer(value):
+        try:
+            submission = ApplicationSubmission.objects.get(id=match[1])
```

Comment:
Maybe we should be looking at using `findall` instead of `finditer` and then doing:

```
referenced_submissions = ApplicationSubmission.objects.filter(id__in=list_of_matches).only('id', 'title)
links = {
    f'#{submission.id}': f'<a href="{submission.get_absolute_url()}">{submission.title} <span class="mid-grey-text">
    for submission in referenced_submissions
}
``` 

in order to get a smaller, single, query for the searches. At a minimum adding `.only` to the existing code is recommended.

Reply:
One query for better than many, fixed that.

Did look up "only" but title is not a real field so not possible to use only('id', 'title'). Title is in form_data and when you have loaded that you have loaded more or less everything so not much to save?